fastplotlib/layouts.py

```diff
@@ -105,32 +103,38 @@ def __init__(
 
             self._controllers[controllers == controller] = controller_types[cam[0]]()
 
+        h = 0
         for i, j in self._get_iterator():
             position = (i, j)
             camera = cameras[i, j]
             controller = self._controllers[i, j]
 
+            if self.names is not None:
+                name = self.names[h]
+                h += 1
+            else:
+                name = None
+
             self.subplots[i, j] = Subplot(
                 position=position,
                 parent_dims=(nrows, ncols),
                 camera=camera,
                 controller=controller,
                 canvas=canvas,
-                renderer=renderer
+                renderer=renderer,
+                name=name
             )
 
-            self.indexers[i, j] = GridPlotIndexer((i,j))
-
         self._animate_funcs: List[callable] = list()
         self._current_iter = None
 
-    def __getitem__(self, index: Union[tuple[int, int], str]):
+    def __getitem__(self, index: Union[Tuple[int, int], str]):
         if type(index) == str:
-            for indexer in self.indexers:
-                if indexer.name == index:
-                    return indexer
+            for subplot in np.concatenate(self.subplots):
```

Comment:
I would use `self.subplots.ravel()` instead since it provides a view whereas `np.concatenate` returns a copy of the array